plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMergeSink.java

```diff
@@ -407,8 +411,8 @@ private Slice writeDeletionVector(
             long rowCount)
     {
         String tablePath = rootTableLocation.toString();
```

Comment:
During my debug for deletion vector case I see that sourcePath is:
```
file:///var/folders/ns/drnmv_551mjcn3rd0ccw1tzm0000gn/T/catalog-dir8727861234911187285/clone_test_dir053n2inpj1/test_clone_merge_cloned_szurs44f36/../clone_merge_deletion_vector_source/part-00000-9f6a8a5a-b5a9-4f09-b7c7-286735267e6b-c000.snappy.parquet
```
which is concatenation of current table and source table. I think it works only because resources are prepared with relative paths. This causes `getReferencedPath` to resolve path as relative and not as file from cloned table. 
You could do something similar what you do to deletion vector path to not use relative paths.